### PR TITLE
Math magicians: events

### PR DESCRIPTION
**Pull request by MohammadYaser**
1. Handle events in a React app.
2. Use React life cycle methods.
3. Understand the mechanism of lifting state up.
4. Use React state.

### CODE REVIEW

File: src/components/Button.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
```
Comment (topeogunleye) [inline src/components/Button.js]: - [ ] Great job using to calculator component to ensure access to the math logic, however, the project requirements ask for `calculate.js` to be imported into the Calculate component. I kindly suggest that you import `calculate.js` into the Calculator component and pass it as a prop to this button component.

image

Remember, you're making good progress! Keep up the good work!

Best regards,
Temitope
Code Reviewer
16/09/2023

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @MohammadYaser, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again


## (Highlights) Good Points: 💪 
🔍 Overall, the student's project fulfills the specified requirements. Here are the highlights and feedback:

Math Logic Files:
✅ The project includes the required "operate.js" and "calculate.js" files, which contain the math logic needed for the Calculator component.

✅ The project effectively handles state with the useState hook, allowing the Calculator component to manage and update its state.

✅ Props are passed correctly to any child components of the Calculate.js component, ensuring seamless communication between parent and child components.

Click Events:
✅ Click events are implemented for all buttons, enabling user interaction with the Calculator component.

Browser Testing:
✅ The code has been tested in the browser, and any errors encountered have been appropriately addressed. Any specific click event implementations that resulted in errors are commented below the relevant code.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- For future pull requests, please notice that your branch name `project-events` does not follow the convention of being named after a feature or action. For example, you could name your branch `added-project-events`. This makes it easier to track the changes and the purpose of each branch.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Button.js (hunk 1 above)
Comment (peterrobert) [reply src/components/Button.js]: This change has been done well

**Review by peterrobert — APPROVED** [review]
Hi @MohammadYaser, 


### Highlights!

[ ] Good feature branch name ✔️
[ ] Descriptive pull request ✔️
[ ] Great work making the changes that has been requested ✔️


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or tag me @peterrobert if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**